src/modm/platform/adc/stm32f3/adc_impl.hpp.in

```diff
@@ -410,3 +434,70 @@ modm::platform::Adc{{ id }}::acknowledgeInterruptFlags(const InterruptFlag_t fla
 	// Writing a zero is ignored.
 	ADC{{ id }}->ISR = flags.value;
 }
+
+/**
+ * @arg slot for the offset register (0..3)
+ * @arg channel channel to which the offset is applied
+ * @arg offset offset value to be applied to the channel
+ * @return true if configuration is successful, false if the ADC is currently converting
+ */
+bool
+modm::platform::Adc{{ id }}::setChannelOffset(const OffsetSlot slot, const Channel channel, const int16_t offset,
+	const bool saturate, const bool enable)
```

Comment:
Does the enable parameter make sense? ie. do you need the channel, offset, saturate info if you disable the slot?
Otherwise you could split this into two methods:
```cpp
enableChannelOffset(OffsetSlot slot, Channel channel, int16_t offset, bool saturate = false);
disableChannelOffset(OffsetSlot slot);
```
Note the defaulted, saturate. That allows you to not have that parameter at all on F3 and H74x/H75x. Then it wouldn't compile if someone set it to true.

```cpp
enableChannelOffset(OffsetSlot slot, Channel channel, int16_t offset);
```